### ISSUE TITLE
SecureStateDriver's unitExistsCache doesn't handle uninteractable units properly

### ISSUE THREAD

**Issue by ls-** [Stanzilla/WoWUIBugs#277]
It got broken all the way back in 9.0.1.36230 when `ShowBossFrameWhenUninteractable` was removed from `unitExistsCache`:

![image](https://user-images.githubusercontent.com/2725970/191057254-46d5b0b7-7ba9-4571-8a82-5ed19a810e77.png)

Encounters where additional uninteractable boss units are present are becoming more and more common, but addons can no longer display them properly.

This doesn't affect the default UI because it relies on
```lua
local function ShouldShowTargetFrame(targetFrame)
	return UnitExists(targetFrame.unit) or ShowBossFrameWhenUninteractable(targetFrame.unit);
end
```
to handle target and boss frames' visibility.

**Comment by Meorawr** [comment]
I'll call it a bug rather than a feature request; I don't think we ever got context on why it was removed to begin with. Does this affect Classic too which - iirc - should have boss frames, but may not have any encounters where this is an actual problem?

**Comment by ls-** [comment]
Yeah, it's definitely a bug. I don't really believe the removal was intentional. I honestly suspect that someone merged something that's based on an older version without rebasing or something like that 😁 
I'm not sure about Classic since I don't play it, but `ShowBossFrameWhenUninteractable` was originally added to `unitExistsCache` in 8.0.1.26629, so I'd assume that it's not an issue in Classic.

**Comment by ls-** [comment]
Looks like it's finally [fixed in 11.2](https://github.com/Gethe/wow-ui-source/blob/ptr2/Interface/AddOns/Blizzard_FrameXML/SecureStateDriver.lua#L58). `ShowBossFrameWhenUninteractable` was replaced with `UnitIsVisible`, at least according to the deprecation docs.
